android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardDownloaderActivity.java

```diff
@@ -384,6 +393,67 @@ public static boolean isCustom(String u) {
     return ret;
   }
 
+  // If a font JSONObject contains multiple font font files, only keep the .ttf source
+  private static void findTTF(JSONObject jsonFont) {
+    boolean updateJsonFont = false;
+    try {
+      JSONArray fontSource = jsonFont.optJSONArray(KMManager.KMKey_FontSource);
+      if (fontSource != null) {
+        boolean containsTTF = hasTTFFont(fontSource);
+        if (containsTTF) {
+          for (int i = fontSource.length() - 1; i >= 0; i--) {
+            String s = fontSource.getString(i);
+            if (!FileUtils.isTTFFont(s)) {
+              updateJsonFont = true;
+              if (Build.VERSION.SDK_INT > 19) {
```

Comment:
Why the two different code paths? Does the `remove` function not work in earlier versions?

Reply:
[remove()](https://developer.android.com/reference/org/json/JSONArray#remove(int)) was added for API 19+ so I was getting exceptions on older devices